src/ploomber/telemetry/telemetry.py

```diff
@@ -242,7 +347,28 @@ def log_api(action, client_time=None, total_runtime=None, metadata=None):
     (telemetry_enabled, uid, is_install) = _get_telemetry_info()
     py_version = python_version()
     docker_container = is_docker()
-    operating_system = get_os()
+    colab = is_colab()
+    if colab:
+        metadata['colab'] = colab
+
+    paperspace = is_paperspace()
+    if paperspace:
+        metadata['paperspace'] = paperspace
+
+    slurm = is_slurm()
+    if slurm:
+        metadata['slurm'] = slurm
+
+    airflow = is_airflow()
+    if airflow:
+        metadata['airflow'] = airflow
+
+    argo = is_argo()
+    if argo:
+        metadata['argo'] = argo
+    if dag and isinstance(dag, DAG):
+        metadata['dag'] = parse_dag(dag)
```

Comment:
maybe to a `try: parse(dag)` and store `None` if it doesn't work. no need to check if it's a DAG and we cover the case where `parse_dag` fails for any reason